java-jacoco/src/main/java/org/sonar/plugins/jacoco/JaCoCoReportMerger.java

```diff
@@ -64,23 +63,22 @@ public static void mergeReports(File reportOverall, File... reports) {
       dataStore.accept((IExecutionDataVisitor) visitor);
     } catch (IOException e) {
       throw new SonarException(String.format("Unable to write overall coverage report %s", reportOverall.getAbsolutePath()), e);
-    } finally {
-      Closeables.closeQuietly(outputStream);
     }
   }
 
   private static boolean loadSourceFiles(ISessionInfoVisitor infoStore, IExecutionDataVisitor dataStore, File... reports) {
-    boolean isCurrentVersionFormat = true;
+    Boolean isCurrentVersionFormat = null;
     for (File report : reports) {
       if (report.isFile()) {
         try {
-          isCurrentVersionFormat = AbstractAnalyzer.readJacocoReport(report, dataStore, infoStore);
+          JacocoReportReader jacocoReportReader = new JacocoReportReader(report).readJacocoReport(dataStore, infoStore);
+          isCurrentVersionFormat = jacocoReportReader.useCurrentBinaryFormat();
         } catch (IOException e) {
           throw new SonarException(String.format("Unable to read %s", report.getAbsolutePath()), e);
         }
       }
     }
-    return isCurrentVersionFormat;
+    return BooleanUtils.isNotFalse(isCurrentVersionFormat);
```

Comment:
@Godin fix in https://github.com/SonarSource/sonar-java/commit/b3457b80f341ee0dc2eb3aebc2e9ffcc44a913fa